### ISSUE TITLE
Goheader check does not work

### ISSUE THREAD

**Issue by kolaente** [golangci/golangci-lint#1446]
I've added goheader to my golangci configuration and it does not seem to work. 

First, I added the template inline to my `.golangci.yml` but it did not report any issues, even though I added some string to my template which I knew wasn't present in at least one of my source code files.

I then thought this could be a weird yaml multi line parsing thing and moved the header configuration to a separate file - still didn't work.

I then added a typo to the path provided for the header configuration file and didn't even complain. Therefore I think that's a bug with golangci rather than goheader.

Ping @denis-tingajkin

---

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.31.0 built from 3d6d0e7 on 2020-09-07T15:14:41Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
run:
  timeout: 1m
  tests: true

linters:
  enable:
    - megacheck
    - govet
    - goconst
    - gocritic
    - gocyclo
    - goerr113
    - goheader
    - gofmt
    - goimports
    - golint
    - misspell
  disable:
    - scopelint # Obsolete, using exportloopref instead
  presets:
    - bugs
    - unused
  fast: false

linter-settings:
  nestif:
    min-complexity: 6
  goheader:
    template-path: code-hesader-template.txt

issues:
  exclude-rules:
    # Exclude some linters from running on tests files.
    - path: _test\.go
      linters:
        - gocyclo
        - deadcode
    - path: pkg/integrations/*
      linters:
        - gocyclo
        - deadcode
        - varcheck
        - unparam
        - bodyclose
    - path: pkg/integrations/*
      text: "unlambda"
      linters:
        - gocritic
    - path: pkg/modules/background/unsplash/unsplash\.go
      linters:
        - bodyclose
    - path: pkg/migration/*
      linters:
        - exhaustive
        - goconst
        - goerr113
    - path: pkg/models/task_collection_filter\.go
      linters:
        - exhaustive
    - path: pkg/utils/random_string\.go
      text: "G404:" # We don't care about cryptographically secure randomness when we're using that utility function.
      linters:
        - gosec
    - path: pkg/modules/dump/*
      linters:
        - goerr113
    - path: pkg/
      text: "err113: do not define dynamic errors, use wrapped static errors instead:"
      linters:
        - goerr113
    - text: "commentFormatting: put a space between `//` and comment text"
      linters:
        - gocritic
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.14.8 linux/amd64
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/kolaente/.cache/go-build"
GOENV="/home/kolaente/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/kolaente/go"
GOPRIVATE=""
GOPROXY="direct"
GOROOT="/nix/store/fkvrggd8q3b543h6bal3clg9arzq6qvq-go-1.14.8/share/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/nix/store/fkvrggd8q3b543h6bal3clg9arzq6qvq-go-1.14.8/share/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="/nix/store/m6h7zh8w6s52clnyskffj5lbkakqgywn-gcc-wrapper-9.2.0/bin/cc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build130339645=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/kolaente/go/src/code.project.io/api /home/kolaente/go/src/code.project.io /home/kolaente/go/src /home/konrad/go /home/kolaente /home /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 28 linters: [asciicheck bodyclose deadcode errcheck exhaustive exportloopref goconst gocritic gocyclo goerr113 gofmt goheader goimports golint gosec gosimple govet ineffassign misspell noctx rowserrcheck sqlclosecheck staticcheck structcheck typecheck unparam unused varcheck] 
INFO [lintersdb] Active presets: [bugs unused]    
INFO [loader] Go packages loading at mode 575 (exports_file|files|imports|types_sizes|compiled_files|deps|name) took 588.27992ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 32.554655ms 
INFO [linters context/goanalysis] analyzers took 19.009978397s with top 10 stages: buildir: 4.553388889s, goimports: 2.177560183s, buildssa: 2.148473581s, misspell: 1.056396913s, gofmt: 810.724203ms, gosec: 800.414228ms, gocritic: 797.338703ms, golint: 575.003889ms, ineffassign: 506.761468ms, unparam: 500.479934ms 
INFO [linters context/goanalysis] analyzers took 2.452251854s with top 10 stages: buildir: 2.126885795s, U1000: 325.366059ms 
INFO [runner] Issues before processing: 128, after processing: 0 
INFO [runner] Processors filtering stat (out/in): path_prettifier: 128/128, identifier_marker: 115/115, exclude: 90/115, filename_unadjuster: 128/128, autogenerated_exclude: 115/128, nolint: 0/3, cgo: 128/128, skip_dirs: 128/128, exclude-rules: 3/90, skip_files: 128/128 
INFO [runner] processing took 19.427772ms with stages: exclude: 6.354159ms, identifier_marker: 4.701155ms, nolint: 4.350988ms, autogenerated_exclude: 1.773076ms, path_prettifier: 1.143783ms, exclude-rules: 600.862µs, skip_dirs: 392.957µs, cgo: 63.351µs, filename_unadjuster: 42.751µs, max_same_issues: 1.29µs, uniq_by_line: 510ns, diff: 420ns, max_from_linter: 390ns, source_code: 370ns, skip_files: 350ns, path_shortener: 310ns, sort_results: 300ns, severity-rules: 280ns, path_prefixer: 260ns, max_per_file_from_linter: 210ns 
INFO [runner] linters took 6.049244704s with stages: goanalysis_metalinter: 4.932070023s, unused: 1.097628265s 
INFO File cache stats: 234 entries of total size 2.2MiB 
INFO Memory: 68 samples, avg is 850.1MB, max is 1419.2MB 
INFO Execution took 6.678034025s                  
```

</details>

**Comment by denis-tingajkin** [comment]
@kolaente Hello, could you attach your .golangci.yml file?

**Comment by ldez** [comment]
@denis-tingajkin the file is in the description ("Config file") :wink:

**Comment by kolaente** [comment]
@denis-tingajkin What @ldez said 🙃

**Comment by denis-tingajkin** [comment]
Oops! Now I see it.

@kolaente 
Is typo in `template-path: code-hesader-template.txt` expected?

**Comment by ldez** [comment]
> I then added a typo to the path provided for the header configuration file and didn't even complain.

**Comment by kolaente** [comment]
Again, what @ldez said :) Looks like I forgot to change that before copying

The code and config is at https://kolaente.dev/vikunja/api by the way, feel free to play around with it.

**Comment by denis-tingajkin** [comment]
@kolaente 
Please retest with the correct linter settings section.

`linter-settings:` --> `linters-settings:`

**Comment by kolaente** [comment]
@denis-tingajkin That seems to have done the trick! Thanks a lot!
